release_bot/releasebot.py

```diff
@@ -124,12 +139,28 @@ def find_newest_release_pull_request(self):
 
             for edge in reversed(edges):
                 cursor = edge['cursor']
-                match = re.match(r'(.+) release', edge['node']['title'].lower())
-                if match and validate(match[1]):
+                last_version = Version(self.github.latest_release())
+                version = ''
+                match = False
+                re_match = re.match(r'(.+) release', edge['node']['title'].lower())
+                if re_match:
+                    match = True
+                    version = re_match[1].strip()
+                elif edge['node']['title'].lower().strip() == "new major release":
+                    match = True
+                    version = str(last_version.next_major())
+                elif edge['node']['title'].lower().strip() == "new minor release":
+                    match = True
+                    version = str(last_version.next_minor())
+                elif edge['node']['title'].lower().strip() == "new patch release":
+                    match = True
+                    version = str(last_version.next_patch())
```

Comment:
The 2 blocks of code you're adding are the same, maybe they can be extracted into a method (e.g. process_version_from_title() - but name it as you like)